apt/risk/data_assessment/dataset_attack_result.py

```diff
@@ -0,0 +1,24 @@
+from dataclasses import dataclass, field
+
+import numpy as np
+
+
+@dataclass
+class DatasetAttackScore:
+    dataset_name: str
+
+
+@dataclass
+class DatasetAttackResult:
+    dataset_name: str
+
+
+@dataclass
+class DatasetAttackScoreWithResult(DatasetAttackScore):
+    result: DatasetAttackResult = field(repr=False)
+
+
+@dataclass
+class DatasetAttackResultPerRecord(DatasetAttackResult):
+    positive_probabilities: np.ndarray
```

Comment:
I stand by my previous comment that I think the two fields positive_probabilities and negative_probabilities are a bit specific and not necessarily representative of any attack that works at the per record level. Maybe we can either call this by a more specific name, or else add another level that only has one array called "per_record_results".

Reply:
I added common methods dealing with the ROC curve and metrics to the DatasetAttackPerRecord class, and they need these positive and negative probabilities. If you think it is too specific, then I should move these too to the DatasetAttackPerRecordKnnProbabilities .